SMHE-1740 object config lijst per sprint - sprint1

natasja-n opened the pull request:
use config service in SetAlarmNotificationsCommandExecutor

https://jenkins.fdp.osgp.cloud/view/SMHE-cucumber/job/gxf-smhe-cucumber-build/133/

natasja-n committed:
SMHE-1740 object config lijst per sprint - sprint1
use config service in SetAlarmNotificationsCommandExecutor

kroesctrl reviewed (changes requested):
Gaarne cucumber testen voor alle protocollen toevoegen. 
Ik vermoed dat het niet gaat werken voor SMR 5.0.0 en 5.1 aangezien ALARM_FILTER_1 alleen maar een [dlmsprofile-smr52.json] zit. We willen dit wel graag in cucumber testen hebben staan.

natasja-n committed:
SMHE-1740 object config lijst per sprint - sprint1
use config service in SetAlarmNotificationsCommandExecutor

natasja-n committed:
SMHE-1740 object config lijst per sprint - sprint1
use config service in SetAlarmNotificationsCommandExecutor

kroesctrl commented on osgp/protocol-adapter-dlms/osgp-dlms/src/main/resources/dlmsprofiles/dlmsprofile-smr43.json:
```diff
@@ -187,6 +187,54 @@
+      ]
+    },
+    {
+      "tag": "ALARM_FILTER_1",
```
Waarom staat ALARM_FILTER_1 hier, deze is identiek aan 4.2.2.
De spec van 4.3 overerft alle specificaties van 4.2.2

natasja-n replied:
Overerving had ik niet goed in de gaten, dit is nu aangepast.

kroesctrl commented on osgp/protocol-adapter-dlms/osgp-dlms/src/main/resources/dlmsprofiles/dlmsprofile-smr52.json:
```diff
@@ -199,6 +199,54 @@
       ]
+    },
+    {
+      "tag": "ALARM_FILTER_1",
```
Overerving van 5.0.0

natasja-n replied:
is aangepast.

kroesctrl commented on osgp/protocol-adapter-dlms/osgp-dlms/src/main/resources/dlmsprofiles/dlmsprofile-smr52.json:
```diff
@@ -199,6 +199,54 @@
+      ]
+    },
+    {
+      "tag": "ALARM_FILTER_2",
```
Ik zie dit object niet in de 5.2 specificatie staan, maar zie het pas in de 5.5 spec. Kun jij aangeven in welk 5.2 document jij dit object hebt gevonden?

kroesctrl replied:
Ik heb inmiddels gevonden dat deze in de 5.2 SagSwell spec staat, dus object hoort inderdaad hier

kroesctrl commented on osgp/protocol-adapter-dlms/osgp-dlms/src/main/resources/dlmsprofiles/dlmsprofile-smr43.json:
```diff
@@ -187,6 +187,54 @@
+          "description": "value",
+          "datatype": "double-long-unsigned",
+          "valuetype": "DYNAMIC",
+          "value": "0",
```
Volgens mij bevat de value geen 0 waarde in de specificatie

kroesctrl commented on osgp/protocol-adapter-dlms/osgp-dlms/src/main/resources/dlmsprofiles/dlmsprofile-dsmr22.json:
```diff
@@ -572,6 +572,30 @@
+          "description": "value",
+          "datatype": "double-long-unsigned",
+          "valuetype": "DYNAMIC",
+          "value": "0",
```
Ik zie in de spec geen 0 waarde staan

natasja-n replied:
Nee, ik heb dat hetzelfde gehouden als bij die Alarm Register, maar ik zal nog wel voor de zekerheid even aan Stefan vragen.

kroesctrl replied:
Zie jij in de spec wel een 0 staan?

natasja-n replied:
Nee, maar dat staat er bij die Alarm register ook niet en zoals ik het begrijp moet je er dezelfde waarden in kunnen zetten. Anders is die ALARM_REGISTER_1 ook niet goed.

kroesctrl replied:
Ik denk dat ook bij AlarmRegister de value niet 0 moet zijn idd

natasja-n replied:
Ik heb dit even met Stefan overlegd omdat ik even de weg kwijt raakte hoe dit nu precies zit met die value. De value wordt nu nog niet gebruikt, maar wordt pas relevant als deze bijvoorbeeld in de simulator gebruikt wordt. Voor nu is daarom de 0 prima. Eigenlijk staat er dan voor nu dus dat er geen alarm in het filter zit en kan dit later aangepast worden.

kroesctrl replied:
Ik ben het er niet mee eens, ik denk dat als we objecten toe voegen aan de configuratie dat deze moet kloppen volgens de specificatie........ We kunnen niet zomaar 0 invullen. Dan moeten we straks nog een keer in de spec duiken

kroesctrl commented on osgp/protocol-adapter-dlms/osgp-dlms/src/main/resources/dlmsprofiles/dlmsprofile-smr500.json:
```diff
@@ -1912,6 +1912,30 @@
+          "description": "value",
+          "datatype": "double-long-unsigned",
+          "valuetype": "DYNAMIC",
+          "value": "0",
```
Ik vraag me af of de value hier 0 moet zijn, ik lees dat niet in de spec

kroesctrl commented:
Ik heb alleen de specificatie maar bekeken......

kroesctrl commented on osgp/protocol-adapter-dlms/osgp-dlms/src/main/resources/dlmsprofiles/dlmsprofile-smr52.json:
```diff
@@ -199,6 +199,54 @@
+          "description": "value",
+          "datatype": "double-long-unsigned",
+          "valuetype": "DYNAMIC",
+          "value": "0",
```
De 0 waarde klopt niet volgens mij

kroesctrl commented on osgp/protocol-adapter-dlms/osgp-dlms/src/main/resources/dlmsprofiles/dlmsprofile-dsmr422.json:
```diff
@@ -1646,6 +1646,30 @@
+          "access": "RW"
+        }
+      ]
     }
```
ALARM_REGISTER_1 zie ik ook in de 4.2.2 spec

natasja-n replied:
is aangepast

kroesctrl commented on osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/alarm/SetAlarmNotificationsCommandExecutor.java:
```diff
@@ -82,20 +83,25 @@ public AccessResultCode execute(
+        this.objectConfigServiceHelper.findOptionalDefaultAttributeAddress(
+            protocol, DlmsObjectType.ALARM_FILTER_1);
+
+    if (alarmFilter1AttributeAddress.isPresent()) {
```
Volgens mij kan deze if eruit....... elk protocol heeft een ALARM_FILTER1

natasja-n replied:
is aangepast

kroesctrl commented on osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/alarm/SetAlarmNotificationsCommandExecutor.java:
```diff
@@ -82,20 +83,25 @@ public AccessResultCode execute(
-            alarmFilter1AttributeAddress,
-            DlmsObjectType.ALARM_REGISTER_1);
+    final Optional<AttributeAddress> alarmFilter1AttributeAddress =
+        this.objectConfigServiceHelper.findOptionalDefaultAttributeAddress(
```
Volgens mij hoef je hier niet findOptionalDefaultAttributeAddress hier kun je ervan uit gaan dat het object bestaat, net als in de oude code

natasja-n replied:
Klopt, is nu geen Optional meer.

kroesctrl commented on osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/alarm/SetAlarmNotificationsCommandExecutor.java:
```diff
@@ -110,8 +116,8 @@ public AccessResultCode execute(
-        this.dlmsObjectConfigService.findAttributeAddress(
-            device, DlmsObjectType.ALARM_FILTER_3, null);
+        this.objectConfigServiceHelper.findOptionalDefaultAttributeAddress(
+            protocol, DlmsObjectType.ALARM_FILTER_3, null);
```
Waarom kies je ervoor om op regel 103-104 de null niet als argument  mee te geven en hier wel in de aanroep van de methode? In de oude code was dit ook identiek

natasja-n replied:
was niet geheel consequent, nu wel.

natasja-n committed:
SMHE-1740 object config lijst per sprint - sprint1
use config service in SetAlarmNotificationsCommandExecutor

Signed-off-by: Natasja <natasja.nortier@alliander.com>

natasja-n committed:
SMHE-1740 object config lijst per sprint - sprint1
SetAlarmNotificationsCommandExecutor, add note in config file when specified in addendum

Signed-off-by: Natasja <natasja.nortier@alliander.com>

natasja-n committed:
Merge branch 'development' into feature/SMHE-1740_configservice_SetAlarmNotificationsCommandExecutor

natasja-n committed:
SMHE-1740 object config lijst per sprint - sprint1
update unit tests

Signed-off-by: Natasja <natasja.nortier@alliander.com>

natasja-n committed:
Merge branch 'development' into feature/SMHE-1740_configservice_SetAlarmNotificationsCommandExecutor

natasja-n committed:
SMHE-1740 object config lijst per sprint - sprint1
remove value = 0 when valuetype is DYNAMIC, add unittest to make sure all fixed in profiles have a value in the configuration

Signed-off-by: Natasja <natasja.nortier@alliander.com>
